Update all ci- containers to reflect main

This is a tracking issue for the process of updating the TVM ci- containers to reflect the following PRs:
- #7926
- #7915

It will also unblock PRs:
- #7858
- #7970 (it's possible we could just include this PR in this update)

Steps:
- [x] Pick the latest revision of `main` and use that specific git hash for following steps. Document it here.
- [x] Build docker containers, tagging them as e.g. `<username>/ci-<container>:v0.<ver>` (we may not need to build all of these if #7970 is not included here)
    - [x] ci-arm
    - [x] ci-cpu
    - [x] ci-gpu
    - [x] ci-i386
    - [x] ci-lint
    - [x] ci-qemu
    - [x] ci-wasm
- [x] Push docker containers to Docker Hub
- [x] Create a draft PR modifying `Jenkinsfile` to point all containers at `<username>/ci-<container>:v0.<ver>`
- [x] Force-push the PR to [ci-docker-staging](https://github.com/apache/tvm/tree/ci-docker-staging) branch
- [x] Jenkins will notice the push and start a build [here](https://ci.tlcpack.ai/job/tvm/job/ci-docker-staging/)
- [x] Debug the build and repeat these steps until the build passes
- [ ] Push the valid containers to `tlcpack/ci-<container>:v0.<ver>`
- [ ] Update the PR to point `Jenkinsfile` to the new containers.
- [ ] Merge the PR.

Let's use this tracking issue to record challenges we face in updating the drivers.

@jroesch can you please note down where you were with this?

cc @tqchen @u99127 @d-smirnov @leandron @tristan-arm

apologies for the light updates here. we determined that the "Frontend : GPU" tests get into a state where either the GPU hardware is inaccessible after a while or TVM's existence check is wrong. Since we didn't change the CUDA version used here--we just updated to 18.04--the theory is that there is some interoperability problem between CUDA running in the containers (at 10.0) and the CUDA driver loaded on the docker host side (either 10.2 or 11.0, depending which CI node you hit).

@tkonolige and I have spent the last couple days running on a test TVM CI cluster using the same AMI (which has only CUDA 11.0). With CUDA 10.0 (ci-gpu) and 11.0 (host), we ran into another similar-looking bug during the GPU unit tests:
```
[ RUN      ] BuildModule.Heterogeneous
[22:11:58] /workspace/src/target/opt/build_cuda_on.cc:89: Warning: cannot detect compute capability from your device, fall back to compute_30.
unknown file: Failure
C++ exception with description "[22:11:58] /workspace/src/runtime/cuda/cuda_device_api.cc:117: 
---------------------------------------------------------------
An error occurred during the execution of TVM.
For more information, please see: https://tvm.apache.org/docs/errors.html
---------------------------------------------------------------
  Check failed: (e == cudaSuccess || e == cudaErrorCudartUnloading) is false: CUDA: no CUDA-capable device is detected
```

We then upgraded ci-gpu to use CUDA 11.0, and this test seemed to pass all the way to the end of the GPU integration tests, modulo a tolerance issue:
```
tests/python/contrib/test_cublas.py::test_batch_matmul FAILED
// ...
>       np.testing.assert_allclose(actual, desired, rtol=rtol, atol=atol, verbose=True)
E       AssertionError: 
E       Not equal to tolerance rtol=1e-05, atol=1e-07
E       
E       Mismatched elements: 2875175 / 3866624 (74.4%)
E       Max absolute difference: 0.00541687
E       Max relative difference: 0.00015383
E        x: array([[[29.647408, 31.88966 , 33.90233 , ..., 34.673954, 32.908764,
E                31.219051],
E               [30.993076, 30.78019 , 33.67124 , ..., 36.1395  , 29.176218,...
E        y: array([[[29.646427, 31.889557, 33.900528, ..., 34.673126, 32.90791 ,
E                31.21726 ],
E               [30.991737, 30.780437, 33.67001 , ..., 36.139397, 29.174744,...
```

we'll try and push this CUDA 11.0 ci-gpu container through the test CI cluster to see how far we can get. feel free to comment if there are concerns updating to CUDA 11.0.

--- a/Jenkinsfile
+++ b/Jenkinsfile
@@ -44,13 +44,13 @@
 //
 
 // NOTE: these lines are scanned by docker/dev_common.sh. Please update the regex as needed. -->
-ci_lint = "tlcpack/ci-lint:v0.62"
-ci_gpu = "tlcpack/ci-gpu:v0.72"
-ci_cpu = "tlcpack/ci-cpu:v0.73"
-ci_wasm = "tlcpack/ci-wasm:v0.70"
-ci_i386 = "tlcpack/ci-i386:v0.72-t0"
-ci_qemu = "tlcpack/ci-qemu:v0.04"
-ci_arm = "tlcpack/ci-arm:v0.03"
+ci_lint = "tlcpack/ci-lint:v0.65"
+ci_gpu = "tlcpack/ci-gpu:v0.74"
+ci_cpu = "tlcpack/ci-cpu:v0.74"
+ci_wasm = "tlcpack/ci-wasm:v0.71"
+ci_i386 = "tlcpack/ci-i386:v0.73"
+ci_qemu = "tlcpack/ci-qemu:v0.05"
+ci_arm = "tlcpack/ci-arm:v0.05"
 // <--- End of regex-scanned config.
 
 // tvm libraries

